mods/creative/inventory.lua

```diff
@@ -194,10 +194,33 @@ function creative.register_tab(name, title, items)
 	})
 end
 
+-- Sort registered items
+local registered_nodes = {}
+local registered_tools = {}
+local registered_craftitems = {}
+
+minetest.register_on_mods_loaded(function()
+	for registered in pairs(minetest.registered_items) do
+		local group = minetest.registered_items[registered].groups or {}
```

Comment:
this is needlessly complicated because iterating will give you `minetest.registered_items[registered]` "for free"
```lua
for name, def in pairs(minetest.registered_items) do
	local group = def.groups or {}
	if ... then
		registered_nodes[name] = def
```

Reply:
It now looks like this:
```
minetest.register_on_mods_loaded(function()
	for name, def in pairs(minetest.registered_items) do
		local group = def.groups or {}

		if group.node or minetest.registered_nodes[name] then
			registered_nodes[name] = def
		end

		if group.tool or minetest.registered_tools[name] then
			registered_tools[name] = def
		end

		if group.craftitem or minetest.registered_craftitems[name] then
			registered_craftitems[name] = def
		end
	end
end)
```

But the buckets will be in the Items and the Tools category. Did I miss something?

Reply:
Use an `elseif` instead?

Reply:
Oh . . . _I knew that_ xD